utility/utils.ts

```diff
@@ -1,3 +1,98 @@
-export function add(a:number, b:number): number {
-    return a + b;
+import * as fsPromise from "fs";
+import { UsersData } from "../model/UsersData";
+import { onlyLettersAndSpacesRegex } from "./constants";
+
+export function validateName(userInput: any): string {
+  if (!userInput.match(onlyLettersAndSpacesRegex)) {
+    throw new Error("Name can't contain digits or special characters!");
+  }
+  userInput = userInput
+    .replace(/\s{2,}/g, " ")
+    .trim()
+    .split(" ");
+  const formattedName = userInput
+    .map((element: string) => {
+      return element.charAt(0).toUpperCase() + element.slice(1).toLowerCase();
+    })
+    .join(" ");
+  return formattedName;
+}
+
+export function validateAge(userInput: any): number {
+  if (isNaN(userInput)) {
+    throw new Error("Kindly enter a numeric value!");
+  }
+  const age = Number(userInput);
+  if (age % 1 != 0) {
+    throw new Error("Do not enter a decimal value");
+  }
+  if (age <= 0 || age >= 100) {
+    throw new Error("Age can only be between 1-99");
+  }
+  return age;
+}
+
+export function validateRollNumber(
+  userInput: any,
+  toDelete: boolean = false
+): number {
+  if (isNaN(userInput)) {
+    throw new Error("Kindly enter a numeric value!");
+  }
+  const rollNumber = Number(userInput);
+  if (rollNumber % 1 != 0) {
+    throw new Error("Do not enter a decimal value or 0");
+  }
+  if (rollNumber <= 0 || rollNumber >= 100000) {
+    throw new Error("Age can only be between 1-99999");
+  }
+  if (toDelete) {
+    return rollNumber;
+  }
+  const data = new UsersData().getUserData();
+  const indexToRemove = data.findIndex(
```

Comment:
Why is the variable named indexToRemove?

Reply:
Changed.